online-fishing-store/src/app/pages/shopping-cart-page/shopping-cart-page.component.ts

```diff
@@ -1,15 +1,59 @@
 import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { ButtonLabel } from 'src/app/enums/button-label-enum';
+import { RouteName } from 'src/app/enums/route-name-enun';
+import { Product } from 'src/app/interfaces/product';
+import { ShoppingCartService } from 'src/app/services/shopping-cart/shopping-cart.service';
 
 @Component({
   selector: 'app-shopping-cart-page',
   templateUrl: './shopping-cart-page.component.html',
   styleUrls: ['./shopping-cart-page.component.scss']
 })
+
 export class ShoppingCartPageComponent implements OnInit {
+  public totalAmount: number;
+  public orderButtonTitle: string = ButtonLabel.ORDER;
+
+  public shoppingCartProducts: Product[];
 
-  constructor() { }
+  constructor(
+    private shoppingCartService: ShoppingCartService,
+    private router: Router
+  ) { }
 
   ngOnInit(): void {
+    this.getShoppingCartProducts();
+    this.getTotalAmount();
+  }
+
+  public getShoppingCartProducts(): void {
+    this.shoppingCartService.getShoppingCartProducts().subscribe((shoppingCartProducts: Product[]) => {
+      this.shoppingCartProducts = shoppingCartProducts;
+    });
+  }
+
+  public removeFromShoppingCart(shoppingCartProduct: Product): void {
+    this.shoppingCartService.removeFromShoppingCart(shoppingCartProduct);
+    this.getTotalAmount(); 
   }
 
+  private getTotalAmount(): void {
+    let totalAmount = 0;
+    this.shoppingCartProducts.forEach((shoppingCartProduct: Product) => {
+      if (!shoppingCartProduct.quantity) shoppingCartProduct.quantity = 1;
```

Comment:
Don't use this structure of "if". When you have data to return use more readable structure of "if else"

If (condition) {
  expression
}

Reply:
Second mistake it is You can't use "return inti forEach"